### PR TITLE
gh-143. Add sessions in sessions Pool

### PR DESCRIPTION
**Pull request by whiteadam2**

### CODE REVIEW

File: apps/monitor-client/lib/api/axios/AxiosSessionsApi.ts [hunk 1]
```diff
@@ -44,4 +44,12 @@ export class AxiosSessionsApi implements IAxiosSessionsApi {
         }
         return null;
     }
+
+    async sync() {
+        const response = await this.http.get(`/sessions_sync`);
+        if (response) {
+            return response.data;
```
Comment (yzhbankov) [inline apps/monitor-client/lib/api/axios/AxiosSessionsApi.ts:51]: data that is returned useless.

File: apps/data-scanner/lib/api/app.mjs [hunk 2]
```diff
@@ -4,7 +4,7 @@ import logger from './logger.mjs';
 import Scanner from './scanner-app/Scanner.mjs';
 
 let http = null;
-let scanner = null;
+export let scanner = null;
```
Comment (yzhbankov) [inline apps/data-scanner/lib/api/app.mjs:7]: you don't need this scanner outside. and you dont need reuse it at all

File: apps/data-scanner/lib/api/rest-api/router.mjs [hunk 3]
```diff
@@ -1,9 +1,12 @@
-import { Router } from 'express';
-import { sessions } from './routes/index.mjs';
+import {Router} from 'express';
+import {scanner} from "../app.mjs";
 
 const router = Router();
 
 router.get('/', (req, res) => res.send('Hello'));
-router.get('/sessions_update', sessions.update);
+router.get('/sessions_sync', async (req, res) => {
```
Comment (yzhbankov) [inline apps/data-scanner/lib/api/rest-api/router.mjs:7]: implement the same logic with makeRequestHandler
add new UseCase for sync and use await appCommand.invoke inside this usecase

File: apps/data-scanner/lib/api/scanner-app/Scanner.mjs [hunk 4]
```diff
@@ -40,6 +40,10 @@ export default class Scanner {
         this.#cronJob.start();
     }
 
+    async sessionsSync() {
+        await appCommand.invoke(COMMANDS.SESSIONS.SYNC, this.#config);
```
Comment (yzhbankov) [inline apps/data-scanner/lib/api/scanner-app/Scanner.mjs:44]: move this to usecase

File: apps/data-scanner/lib/models/TelegramClientBase.mjs [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import {TelegramClient} from "telegram/index.js";
+
+export class TelegramClientBase extends TelegramClient {
+
+    #config = null;
+
+    constructor(session) {
+        super(session, TelegramClientBase.config.apiId, TelegramClientBase.config.apiHash, {});
+        this.#config = TelegramClientBase.config;
+        console.log(this.#config);
```
Comment (yzhbankov) [inline apps/data-scanner/lib/models/TelegramClientBase.mjs:10]: remove console

File: apps/data-scanner/lib/models/TelegramClientBase.mjs [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import {TelegramClient} from "telegram/index.js";
```
Comment (yzhbankov) [inline apps/data-scanner/lib/models/TelegramClientBase.mjs:1]: " --> '

File: apps/data-scanner/lib/models/sessions/Client.mjs [hunk 7]
```diff
@@ -1,5 +1,5 @@
-import { TelegramClientAdapter } from './TelegramClientAdapter.mjs';
-import { Api } from 'telegram';
```
Comment (yzhbankov) [inline apps/data-scanner/lib/models/sessions/Client.mjs:2]: setup you linter, do not remove spaces

File: apps/data-scanner/lib/models/sessions/ClientsPool.mjs [hunk 8]
```diff
@@ -171,13 +172,13 @@ export class ClientsPool {
                                 checkDate: new Date().toISOString(),
                             });
                         } else
-                            await new Contact().updateTrackedStatus({ trackedPhone: '+' + user.phone, tracked: false });
```
Comment (yzhbankov) [inline apps/data-scanner/lib/models/sessions/ClientsPool.mjs:174]: do not remove spaces in all files

File: apps/monitor-client/lib/api/mock/MockSessionsApi.ts [hunk 9]
```diff
@@ -4,6 +4,7 @@ export interface IMockSessionsApi {
     readList: (params: ReadParams) => Promise<null | {}>;
     add: (params: FormData) => Promise<null>;
     del: (id: string) => Promise<null>;
+    syncWithPool: () => Promise<null>;
```
Comment (yzhbankov) [inline apps/monitor-client/lib/api/mock/MockSessionsApi.ts:7]: just sync

File: apps/data-scanner/config/config-schema.json [hunk 10]
```diff
@@ -1,38 +1,114 @@
...
+        ],
+        "cron": [
+          "required",
+          "string"
+        ],
+        "apiId": [
+          "required",
+          "positive_integer"
+        ],
+        "apiHash": [
```
Comment (yzhbankov) [inline apps/data-scanner/config/config-schema.json:107]: revert formatting

File: shared/env/.env.defaults [hunk 11]
```diff
@@ -47,4 +47,4 @@ IN_CONTACT_UI_SERVICE_NAME=incontact-ui
 IN_CONTACT_UI_LIMIT_TRACKED_PHONE=5
 
 ######## BOT ########
-IN_CONTACT_BOT_TOKEN=YOUR_TELEGRAM_BOT_TOKEN
+IN_CONTACT_BOT_TOKEN=5161548178:AAFuD_WIBjzRvICrnxME3lMxNyjECz_xwPs
```
[the remark this answers is not in the document]
Comment (whiteadam2) [reply shared/env/.env.defaults:51]: done

File: apps/monitor-client/lib/api/mock/MockSessionsApi.ts (hunk 9 above)
Comment (whiteadam2) [reply apps/monitor-client/lib/api/mock/MockSessionsApi.ts:7]: done

File: apps/data-scanner/config/config-schema.json (hunk 10 above)
Comment (whiteadam2) [reply apps/data-scanner/config/config-schema.json:107]: done

File: apps/data-scanner/lib/usecases/UpdateSessionsPool.mjs [hunk 12]
```diff
@@ -8,5 +8,6 @@ export class UpdateSessionsPool {
 
         const removeList = await new Contact().getList({ tracked: false, withSession: true });
         await ClientsPool.removeContacts(removeList);
+        console.log(ClientsPool.pool.map((client) => client.sessionId));
```
Comment (yzhbankov) [inline apps/data-scanner/lib/usecases/UpdateSessionsPool.mjs:11]: remove logs

**Review by yzhbankov — APPROVED** [review]

File: apps/data-scanner/lib/usecases/UpdateSessionsPool.mjs (hunk 12 above)
Comment (whiteadam2) [reply apps/data-scanner/lib/usecases/UpdateSessionsPool.mjs]: done

**Pull request merged**